src/components/Modals/Send/hooks/useSendDetails/useSendDetails.tsx

```diff
@@ -154,15 +154,20 @@ export const useSendDetails = (): UseSendDetailsReturnType => {
     }
   }, [accountSpecifier, chainAdapterManager, contractAddress, getValues, wallet])
 
-  const handleNextClick = async () => {
-    try {
-      setLoading(true)
-      history.push(SendRoutes.Confirm)
-    } catch (e) {
-      moduleLogger.error(e, 'This should never happen')
-    } finally {
-      setLoading(true)
+  const debouncedEstimateFormFees = useMemo(() => {
+    return debounce(estimateFormFees, 1000, { leading: true })
+  }, [estimateFormFees])
+
+  // Stop calls to debouncedEstimateFormFees on unmount
+  useEffect(() => {
+    return () => {
+      debouncedEstimateFormFees.cancel()
     }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
I think it works either way, as it is an unmount cleanup function, so it runs when the component unmounts to stop the active debouncing process.

Though, I don't see any reason why we need to disable the exhaustive hooks. This will return the same cleanup function with the dependency there.

I added the dependency and removed the comment.